File: Github/TallerThreePython/Rhomb.py
'''
Created on 25/11/2017

@author: Stiven Caceres Hernandez
'''
import random

class main():
    matrix = []
    print("Welcome")
    print("This program will print the even, odd, positive and")
    print("negative numbers of a matrix m * m with random numbers.\n")
    print("write the value of \"m\"\n")
    size = input()
    size = int(size)
    matrix = []
    board = []
    #create matrix
    for i in range(size):
        matrix.append([0]*size)
        board.append([0]*size)
    #fill matrix
    for i in range(0,size):
        for j in range(0,size):
            matrix[i][j] = random.randint(10,99)
    #print matrix
    print ("first matrix\n")
    for i in matrix:
        print (i)
    
    firstValue = int((size - 1) / 2)    #principle of the diamond
    half = True                         #if it's half the diamond
    
    #if the size of the matrix is even or odd
    if size % 2 ==0:
        finalyValue = firstValue + 1
    else:
        finalyValue = firstValue
        
    #fill new matrix
    for i in range(firstValue):
        print(firstValue)
        for j in range(finalyValue):
            board[i][j] = matrix[i][j]
            
        if half == True:
            if size % 2 == 0:
                if (firstValue - 1) >= 0 and finalyValue < size:
                    firstValue = firstValue - 1
                    finalyValue = finalyValue + 1
                    print(firstValue)
                else:
                    half = False
                    firstValue = 0
                    finalyValue = size - 1
            else:
                if (firstValue - 1) > 0 and finalyValue < size:
                    firstValue = firstValue - 1
                    finalyValue = finalyValue + 1
                else:
                    half = False
                    firstValue = 0
                    finalyValue = size - 1
        else:
            firstValue = firstValue + 1
            finalyValue = finalyValue -1
    
    print("\nnew matrix\n")        
    for i in board:
        print (i)

File: Github/TallerThreeJava/src/Rhomb.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Random;

public class Rhomb {
	
	//these are the objects to use
	static BufferedReader br = new BufferedReader (new InputStreamReader (System.in));
	static BufferedWriter bw = new BufferedWriter (new OutputStreamWriter (System.out));
	static Random rd = new Random();
			
	public static void main (String []args) throws IOException
	{
		//all variables
		int m, firstValue, finalyValue,i,j;
		boolean half;
		int[][] board, boardRhomb;
		
		//Welcome message
		bw.write("Welcome\n"
				+ "This program will print the numbers of a matrix\n"
				+ "\"m * m\" only those that are inside a rhombus\n\n"
				+ "write the value of \"m\"\n");
		bw.flush();
		m = Integer.parseInt(br.readLine());
		
		board = new int [m][m];
		boardRhomb = new int [m][m];
		
		bw.write("\nfirst matrix\n");
		bw.flush();
		//fill matrix with numbers random
		for (i = 0; i < m; i++)
		{
			bw.write("\n");
			bw.flush();
			for(j = 0; j < m; j++)
			{
				board[i][j] = rd.nextInt(100);
				bw.write("["+board[i][j]+"]\t");
				bw.flush();
			}
		}
		
		firstValue = (m - 1) / 2;	//principle of the diamond
		half = true;				//if it's half the diamond
		
		//if the size of the matrix is even or odd
		if(m % 2 == 0)
		{
			finalyValue = firstValue + 1;
		}
		else
		{
			finalyValue = firstValue;
		}
		
		bw.write("\n\nMatrix Rhomb\n");
		bw.flush();
		
		//fill the matrix of the rhomb
		for (i = 0; i < m; i++)
		{
			for(j = firstValue; j <= finalyValue; j++)
			{
				boardRhomb[i][j] = board[i][j];
			}
			if (half == true)
			{
				if (m % 2 == 0)
				{
					if((firstValue - 1) >= 0 && finalyValue < m)
					{
						firstValue -= 1;
						finalyValue += 1; 
					}
					else
					{
						half = false;
						firstValue = 0;
						finalyValue = m-1;
					}
				}
				else
				{
					if((firstValue - 1) > 0 && finalyValue < m)
					{
						firstValue -= 1;
						finalyValue += 1; 
					}
					else
					{
						half = false;
						firstValue = 0;
						finalyValue = m-1;
					}
				}
			}
			else
			{
				firstValue += 1;
				finalyValue -= 1;
			}
		}
		
		//print the new matrix
		for(i = 0; i < m; i++)
		{
			bw.write("\n");
			bw.flush();
			for (j = 0; j < m; j++)
			{
				bw.write("[" + boardRhomb[i][j] + "]\t");
				bw.flush();
			}
		}
	}
}
